### PR TITLE
[IMP] website: events redesign

### PR DESCRIPTION
**Pull request by morm-odoo**
This PR improves the overall design of all the views in the Frontend
side of the module Event.
The design was outdated and was not consistent between the different
pages.

List of changes:
- events: cleaning of scss, filters consistency and responsive
(with off-canvas).
- submenu: design and responsive behavior (dropdown)
- registration page : overall design and the ticket list is inside a
modal, to help with cases like having a lot of tickets. To be improved,
on a separate page instead of a modal.
- talks and agenda: design, cleaning of scss, filters consistency and
responsive (with off-canvas).
- sponsor template: minor design modification
- exhibitors: design, cleaning of scss, filters consistency and responsive
(with off-canvas).
- location: minor design modification to be consistent with others pages,
but need improvement in the near future.
- meet: design, cleaning of scss
- booth selection: overall design

task-3083658

### CODE REVIEW

File: addons/website_event/views/event_templates_page_registration.xml
Comment (thiryr) [inline addons/website_event/views/event_templates_page_registration.xml:362]: You issue with the min/max comes from here. See before this was located inside this block `<t t-if="len(event.event_ticket_ids) &gt; 1">`

So in the override, we knew the block we were replacing was only there if we indeed had tickets. But not anymore.

Even if it's a modal, it's still going to get rendered before being displayed. I would say we can simply keep the behavior here. If there is not more than one ticket, don't show the range as it is not relevant. Using the same t-if condition as below for this div.
```suggestion
                <div t-if="len(event.event_ticket_ids) &gt; 1" class="o_wevent_price_range ms-2"/>
```

File: addons/website_event/static/tests/tours/tickets_questions.js
Comment (thiryr) [inline addons/website_event/static/tests/tours/tickets_questions.js:12]: Here indeed you will want to add a step where you click on the register button that opens the new modal.

Then keep this step to check the actual 'register' button in the modal is disabled. And globally readjust the tour so that the steps match the new layout, but I believe most of the tour should still work since it tests the other modal and some pages you haven't touched yet afaict

**Review by stefanorigano — CHANGES_REQUESTED** [review]
Thanks @morm-odoo , almost there 👍 
Some comments for you in the code and some remarks here:

- In the "Register" page, remove navbar's [Register] button, it's not necessary. 
image

- Cards' footers looks like dividers in listView 
image

- In listView, images border-radius is wrong
image

- As discussed, you should review the controllers design, especially on medium res.
image

- "Agenda" page: search bar design is wrong, it should use  `input-group`
image

- "Agenda" page: the layout uses `container-fluid` and goes outside the wrapper. It should use  `container` instead. Beside solving the design issue, using the right class ensures that the layout adapts according to user's preferences  ( "Theme" tab > "Page Layout" controller)
image

File: addons/website_event/views/event_templates_page.xml
Comment (thiryr) [inline addons/website_event/views/event_templates_page.xml:36]: To hide this when on the '/register' page, you can just add another condition. This time I would actually do it from the route like we originally did for the others.

So here add `and not hide_register_cta` and in the `def _prepare_event_register_values` add an entry `'hide_register_cta': True`

**Review by thiryr — COMMENTED** [review]
:wave:

File: addons/website_event/views/event_templates_list.xml
Comment (thiryr) [inline addons/website_event/views/event_templates_list.xml:74]: I guess we should match the indent here

**Review by thiryr — COMMENTED** [review]
Didn't end up giving it a full tour on friday but here's a few comments.
Found a nice 2-liner for the menu name issue too :)

Will keep looking at this monday, cheers

File: addons/website_event/views/event_templates_page_registration.xml
Comment (thiryr) [inline addons/website_event/views/event_templates_page_registration.xml:136]: So, this is the only part we actually show on mobile right?
I'd add a comment to specify that, it's not super obvious without being familiar with `d-lg-none`

File: addons/website_event_track/static/src/scss/event_track_templates_online.scss
Comment (thiryr) [inline addons/website_event_track/static/src/scss/event_track_templates_online.scss:55]: :thinking:

File: addons/website_event_exhibitor/views/event_exhibitor_templates_list.xml
Comment (thiryr) [inline addons/website_event_exhibitor/views/event_exhibitor_templates_list.xml:88]: Can be collapsed into the label

File: addons/website_event_booth/views/event_booth_templates.xml
Comment (thiryr) [inline addons/website_event_booth/views/event_booth_templates.xml:122]: :thinking:

**Review by thiryr — COMMENTED** [review]
Mostly trying to reduce the indentations.
Overall the structure seems ok, a lot of stuff moving so I'll go over it commit by commit and check that every variable is theoretically accessible but I don't think there will need to be any big change :)

**Review by thiryr — COMMENTED** [review]
Some more to fix the tests

File: addons/website_event/views/event_templates_page.xml
Comment (stefanorigano) [inline addons/website_event/views/event_templates_page.xml:19]: Look s a bit too "crowded"... I'd suggest to use `my-3` everytime.
Beside making breath the layout, it would also be more consistent with the rest (notably the index page navbar).

| _ |
|--------|
| **This PR** |
| image |
| **Suggestion** |
| image |

**Review by stefanorigano — CHANGES_REQUESTED** [review]
Hello @morm-odoo , almost there 💪
Some remarks for you in the code.

We should finalize, could you:
- revolve pr's feedbacks you have already addressed 
- clean the branch (squash commits, write proper commit/pr descriptions)
- add task-ID


Also, somehow related to your changes: The agenda HTML structure is off, seems like the amount of `<tr>` rendered on screen is wrong, could you have a look a that? 

| Agenda (Design Fair Los Angeles) |
|--------|
| **Master** |
| image |
| **This PR** |
| image |

File: addons/website_event_track/static/src/scss/event_track_templates_online.scss
Comment (morm-odoo) [reply addons/website_event_track/static/src/scss/event_track_templates_online.scss:56]: Added a todo comment, to check after if it's still necessary, as it's breaking the layout

File: addons/website_event/views/event_templates_page_registration.xml
Comment (morm-odoo) [reply addons/website_event/views/event_templates_page_registration.xml:56]: line 45 in left side

**Review by stefanorigano — COMMENTED** [review]
Thanks @morm-odoo , almost there 👍 
I pushed some minors on top, but just minors... follows the list of the changes applied. 

c886f16728d7c49b68f3a7d5af5cbf9b3d5a80f0: fix - border is rendered but not visible 

| Before | After |
|--------|--------|
| image | image | 

72b8dfecc91270713bf3bc6b3426903601fab4aa : fix - slightly adapt grid's gap depending by the resolution

dde6d185887a22fa1769c6b501abca20a856d955 : allows "configure tickets" link classes adaptation

12cd88c33287d02d39b8660a3fdf45018f15ceae: avoid that content and footer touch

de6dd92f1bd7909e7368fb456aff2ae2bb988036: badges design adaptations + fallback for no-color


cd3c6245ab86a4fb9b119be98a44282d5d133094: addressing minor layout issue on SM

| Before  | After |
|--------|--------|
| image | image |

File: addons/website_event_exhibitor/views/event_exhibitor_templates_list.xml
Comment (thiryr) [inline addons/website_event_exhibitor/views/event_exhibitor_templates_list.xml:61]: Here's an idea. We can use the option templates so it's much less monolithic and we respect the options of the users.
See this here commit, which seems to work fine.
https://github.com/odoo-dev/odoo/commit/bc74708de04601f6cc1afb7e59e1236c9c0fb07e

On that note the options don't appear anymore because we used 'nav' as a selector ^^
Easy fix here, no need to use 'nav' there's only one of these. The options won't be available when using the editor on mobile.. but the editor on mobile doesn't really work anyway, and nobody would use it even if it did probably.
https://github.com/odoo-dev/odoo/commit/651a65fe11bf7f810f9a236dc7932cd5f9336e32

**Review by thiryr — COMMENTED** [review]
A suggestion and a few comments :)

**Comment by tde-banana-odoo** [comment]
@stefanorigano Is it validated on your side ?

**Review by stefanorigano — APPROVED** [review]
> @stefanorigano Is it validated on your side ?

Hello @tde-banana-odoo , yes, design is good 👍 

I got a `service-worker.js` error in the latest runbot build tho.
IMO it's unrelated to this PR, I'm reporting here just to be sure.
```console
service-worker.js:251     Uncaught (in promise) TypeError: Failed to fetch
    at processFetchRequest (service-worker.js:251:30)
    at service-worker.js:356:23
```

image

**Comment by Xavier-Do** [comment]
Upgrade exception [#260](https://runbot.odoo.com/web/#id=260&view_type=form&model=runbot.upgrade.exception) added.

Waiting the forward-port of this pr, the exception will be applied on all builds, and create an inconsistent state for migrations.
**Please forward port this asap up to master without change.** Exception should be forward-ported before the end of the week.

If you need to apply any change before it reaches master, please notify runbot team.

Details:
```
view:website_event_sale.modal_ticket_registration
view:website_event.modal_ticket_registration
view:website_event.navbar
```
cc @KangOl @nseinlet

File: addons/website_event_track/static/src/scss/event_track_templates_online.scss
Comment (thiryr) [reply addons/website_event_track/static/src/scss/event_track_templates_online.scss:56]: For reference, this was introduced for youtube live chat to display properly

@morm-odoo  I tested it and this does seem to make the chat unusable on smaller screens (basically anything smaller than a laptop screen horizontally). Some buttons gradually disappear and we can't really read the chat.

Is there any way we can keep this without breaking your design?
Maybe we can add a class conditionally only when the youtube chat is available?

If difficult to fix I guess it can be merged and this handled with a bugfix but it would be best to handle it now.

**Review by thiryr — COMMENTED** [review]
Hey @morm-odoo, did a final pass since this shouldn't change much anymore.

Seems fine overall, just a few cleanup comments. But one annoying issue that we put on the side I'd like to fix before merging if possible.

The traceback indeed seems unrelated as far as I could tell

**Review by thiryr — APPROVED** [review]
Seems ok, thanks for the changes :)

**Comment by thiryr** [comment]
@tde-banana-odoo

**Comment by tde-banana-odoo** [comment]
@robodoo r+ rebase-merge

**Comment by Xavier-Do** [comment]
I would have prefer to wait for the upgrade pr to be merged but since it doesn't move much: 

Upgrade exception [#262](https://runbot.odoo.com/web/#id=262&view_type=form&model=runbot.upgrade.exception) added.

Waiting the forward-port of this pr, the exception will be applied on all builds, and create an inconsistent state for migrations.

Please merge the corresponding upgrade pr asap

Details:
```
view:website_event_sale.registration_template
view:website_event.fold_register_details
```
cc @KangOl @nseinlet

**Comment by Xavier-Do** [comment]
@robodoo r+

**Comment by tde-banana-odoo** [comment]
> @robodoo r-
> 
> Upgrade is not merged yet, so this blows up: https://runbot.odoo.com/runbot/build/53103486

Keep your hands off my PRs you sneaky monkey ! I challenge you to a Rocher Suchard battle !

**Comment by Xavier-Do** [comment]
> > @robodoo r-
> > Upgrade is not merged yet, so this blows up: https://runbot.odoo.com/runbot/build/53103486
> 
> Keep your hands off my PRs you sneaky monkey ! I challenge you to a Rocher Suchard battle !

Added an exception to make it pass but as you wish

**Pull request closed without merging**
